geoproxy/main.go

```diff
@@ -750,29 +764,38 @@ func getResponse(id uuid.UUID) PostList {
 
 func serveResponseThenCleanup(id uuid.UUID) {
 	if requestMutex, ok := requestMutexMap[id]; ok {
-		fmt.Println("found mutex")
 		requestMutex.Lock() // We lock here in case we have two or more responses arrive after timeout
 		// We don't want both responses triggering us to serve the response.
 		// This shouldn't be a problem if all responses arrive in a timely manner though
 		defer requestMutex.Unlock()
 
-		var responseEntries = getResponse(id)
+		// Get the response writer
+		var resWriter = responseWriterMap[id]
+		var pl = getResponse(id)
```

Comment:
TODO: I think we should check if the id key still exists in queryMap (just in case)